app/views/RoomInfoView/index.js

```diff
@@ -214,7 +214,7 @@ class RoomInfoView extends React.Component {
 		}
 
 		const permissions = await RocketChat.hasPermission([editRoomPermission], room.rid);
-		if (permissions[0] && !room.prid) {
+		if (permissions[0] && (!room.prid || this.t === 'p')) {
```

Comment:
By reading the code, now I got your point.
You can just remove the condition.
If the user has the permission, he can edit any channel type: `p`, `c`, `d`, `l`.
The extra condition was just because discussions should be changed back when it was first implemented.

```suggestion
		if (permissions[0]) {
```